### PR TITLE
ASM-5858 VSAN Teardown Work

### PR DESCRIPTION
**Pull request by sushilrai**
Enable disk-group deletion of servers added to the VSAN enabled cluster.

### CODE REVIEW

File: lib/puppet/provider/vc_vsan_disk_initialize/default.rb [hunk 1]
```diff
@@ -8,16 +8,30 @@
...
+    end
+  end
+
+  def host_system(host)
+    cluster_hosts.find {|x| x.name == host}
   end
 
   def exists?
-    false
+    resource[:ensure] == :present ? false : true
```
Comment (gavin-scott) [inline lib/puppet/provider/vc_vsan_disk_initialize/default.rb]: I would update this to `resource[:ensure] != :present`.

File: lib/puppet/provider/vc_vsan_disk_initialize/default.rb [hunk 2]
```diff
@@ -42,23 +56,27 @@ def cluster_hosts
...
                                :ns => 'urn:vim25',
                                :ssl => true,
                                :rev => "6.0",
                                :path => '/vsanHealth' )
-    @hsconn.cookie = vim.cookie
-    @hsconn.debug = vim.debug
-    @hsconn
+    conn.cookie = vim.cookie
+    conn.debug = vim.debug
+    conn
```
Comment (gavin-scott) [inline lib/puppet/provider/vc_vsan_disk_initialize/default.rb]: Looks like this is used in at least two different places. Do you specifically want to create a new connection for both? Or would it be better to memoize this? (`@hsconn ||= ...`)

**Comment by gavin-scott** [comment]
A couple of minor comments and I'm not familiar with the VSAN APIs but overall looks OK to me.

File: lib/puppet/provider/vc_vsan_disk_initialize/default.rb (hunk 2 above)
Comment (sushilrai) [inline lib/puppet/provider/vc_vsan_disk_initialize/default.rb]: Same connection is request is not working when we are making simultaneous calls. That's why I have change the code.

Comment (gavin-scott) [inline lib/puppet/provider/vc_vsan_disk_initialize/default.rb]: Fair enough, thanks.

**Pull request merged**